### PR TITLE
388 - Adding coverage to pytest-pixl

### LINKED ISSUE #388: Track coverage of pytest-pixl
### Definition of Done / Acceptance Criteria

pytest-pixl plugin coverage is tracked by pytest-cov. Needs some [extra fun implementation](https://pytest-cov.readthedocs.io/en/latest/plugins.html), expect that it should be basically ~100%~ 90% covered

### PR DESCRIPTION
**Pull request by mxochicale**
## Description
Fixes #388: Adding coverage to `pytest-pixl`, including the following tests to cover _create_default_json and FtpHostAddress:
* test_create_default_json_file 
* test_ftp_host_address

See [report](https://app.codecov.io/gh/SAFEHR-data/PIXL/tree/388-coverage-of-pytest-pixl/pytest-pixl%2Fsrc%2Fpytest_pixl) in codecov

## Type of change
Please delete options accordingly to the description.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Suggested Checklist

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have passed on my local host device. (see further details at the [CONTRIBUTING](https://github.com/SAFEHR-data/PIXL/blob/main/CONTRIBUTING.md#local-testing) document)
- [x] Make sure your branch is up-to-date with main branch. See [CONTRIBUTING](https://github.com/SAFEHR-data/PIXL/blob/main/CONTRIBUTING.md) for a general example to syncronise your branch with the `main` branch.
- [x] I have requested PR review to `UCLH-Foundtry/arc-dev`
- [x] I have adddressed and marked as resolved all the review comments in my PR.
- [x] Finally, I have selected `squash and merge`

### CODE REVIEW

File: pytest-pixl/README.md [hunk 1]
```diff
@@ -8,6 +8,11 @@ Installable `pytest` plugin module providing common test fixtures used throughou
 python -m pip install -e ../pixl_core -e ".[test]"
 ```
 
+## pytest-cov’s engine
+```
+COV_CORE_SOURCE=src COV_CORE_CONFIG=.coveragerc COV_CORE_DATAFILE=.coverage.eager pytest --cov=src --cov-append
```
Comment (stefpiatek) [inline pytest-pixl/README.md:13]: do we need to set these variables when running github actions?

File: pytest-pixl/tests/test_ftpserver_fixture.py [hunk 2]
```diff
@@ -11,11 +11,32 @@
...
+
 import pytest
+from conftest import TEST_DIR
+from pytest_pixl.dicom import _create_default_json
+from pytest_pixl.plugin import FtpHostAddress
 
 
-@pytest.mark.pytester_example_path("tests/samples_for_fixture_tests/test_ftpserver_fixture")
+@pytest.mark.pytester_example_path(
+    str(TEST_DIR) + "/samples_for_fixture_tests/test_ftpserver_fixture"
```
Comment (stefpiatek) [inline pytest-pixl/tests/test_ftpserver_fixture.py:24]: FYI this is a bit cleaner IMHO
```suggestion
    f"{TEST_DIR}/samples_for_fixture_tests/test_ftpserver_fixture"
```

**Review by stefpiatek — COMMENTED** [review]
This is looking good, but we're still not getting the pytest pixl plugin as having full coverage in CI. You can check it out in sentry to see what has coverage there - I expect its setting up the github action correctly that is needed

File: pytest-pixl/tests/test_ftpserver_fixture.py (hunk 2 above)
Comment (mxochicale) [reply pytest-pixl/tests/test_ftpserver_fixture.py]: Thanks, sorted out here [c3e4e5f](https://github.com/SAFEHR-data/PIXL/pull/501/commits/c3e4e5fd68a19b1ef3c2303005ed097c73b35f03)

File: pytest-pixl/README.md (hunk 1 above)
Comment (mxochicale) [reply pytest-pixl/README.md]: Well spotted. Just sorted out here [85b7ff7](https://github.com/SAFEHR-data/PIXL/pull/501/commits/85b7ff74e21e68a5d7a48690250716cb400cb9ab)

**Comment by mxochicale** [comment]
Thanks Stef, let me know your thoughts! Coverage [report](https://app.codecov.io/gh/SAFEHR-data/PIXL/tree/388-coverage-of-pytest-pixl/pytest-pixl%2Fsrc%2Fpytest_pixl) for dicom.py and plugin.py seem to be >90%

**Review by stefpiatek — APPROVED** [review]
Ah perfect, now we're tracking the actual running of the plugin 🎉 

Thanks for getting this in Miguel

**Pull request merged**